docs/ml-clustering.md

```diff
@@ -104,4 +104,31 @@ Refer to the [Java API docs](api/java/org/apache/spark/ml/clustering/LDA.html) f
 {% include_example java/org/apache/spark/examples/ml/JavaLDAExample.java %}
 </div>
 
-</div>
\ No newline at end of file
+</div>
+
+## Bisecting k-means
+
+Bisecting K-means can often be much faster than regular K-means, but it will generally produce a different clustering.
+
+Bisecting k-means is a kind of [hierarchical clustering](https://en.wikipedia.org/wiki/Hierarchical_clustering) in the fashion of
+division: all observations start in one cluster, and splits are performed recursively as one moves down the hierarchy.
```

Comment:
Exchange the two lines above, it's better to give the overview of Bisecting k-means firstly.